modin/engines/base/frame/data.py

```diff
@@ -1241,7 +1241,7 @@ def _map_reduce(self, axis, map_func, reduce_func=None):
         )
         return self._compute_map_reduce_metadata(axis, reduce_parts)
 
-    def _map(self, func, dtypes=None, validate_index=False, validate_columns=False):
```

Comment:
As I understand it, these changes are basically a subset of the changes in #2744 applied because of performance issues - does that sound right? I just want to make sure that I'm on the same page.